libs/types-common/src/Data/Id.hs

```diff
@@ -87,54 +87,62 @@ import Servant (FromHttpApiData (..), ToHttpApiData (..))
 import Test.QuickCheck
 import Test.QuickCheck.Instances ()
 
-data IdTag = A | C | I | U | P | S | T | STo
+data IdTag
+  = Asset
+  | Conversation
+  | Invitation
+  | User
+  | Provider
+  | Service
+  | Team
+  | ScimToken
 
 idTagName :: IdTag -> Text
-idTagName A = "Asset"
-idTagName C = "Conv"
-idTagName I = "Invitation"
-idTagName U = "User"
-idTagName P = "Provider"
-idTagName S = "Service"
-idTagName T = "Team"
-idTagName STo = "ScimToken"
+idTagName Asset = "Asset"
+idTagName Conversation = "Conv"
```

Comment:
Kind of annoying having this to basically replace a `Show` instance, now after the refactor, but don't want to pile up more stuff for you to do 👍